torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -1165,14 +1188,23 @@ struct to_ir {
     // and (2) only enable this OPTIONAL_NONE when loading newer
     // graphs because it is incompatible with older graphs.
     // Refinement none(name, RefinementKind::OPTIONAL_NONE);
-    if (auto optional_type = lhs_value->type()->cast<OptionalType>()) {
+    if (const auto optional_type = lhs_value->type()->cast<OptionalType>()) {
       Refinement present(name, optional_type->getElementType());
       if (tok == TK_IS) {
         return RefinementSet({}, {present});
       } else { // TK_ISNOT
         return RefinementSet({present}, {});
       }
     }
+    if (const auto union_type = lhs_value->type()->cast<UnionType>()) {
+      UnionTypePtr remaining = union_type->withoutNone();
```

Comment:
Do we have any other callers of `withoutNone` ? This seems like a very specific API to add to Union Types if this is the only caller. I would expect this to just be something like `union_type->subtractTypeSet(None)`

Reply:
Fixing in previous PR